wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,186 @@
+            auditReport = createTempDirectory(null).resolve("report_" + dateTime + ".txt");
+            Files.createFile(auditReport);
+
+            CodenvyLicense license = null;
+            try {
+                license = licenseManager.load();
+            } catch (LicenseException ignored) {
+                //Continue printing report without license info
+            }
+            reportPrinter.printHeader(auditReport, userManager.getTotalCount(), license);
+            printAllUsers(auditReport);
+        } catch (Exception exception) {
+            if (auditReport != null) {
+                deleteReportDirectory(auditReport);
+            }
+            LOG.error(exception.getMessage(), exception);
+            throw new ServerException(exception.getMessage(), exception);
+        } finally {
+            inProgress.set(false);
+        }
+
+        return auditReport;
+    }
+
+    private void printAllUsers(Path auditReport) throws ServerException {
+        Page<UserImpl> currentPage = userManager.getAll(30, 0);
+        do {
+            //Print users with their workspaces from current page
+            for (UserImpl user : currentPage.getItems()) {
+                List<WorkspaceImpl> workspaces;
+                int permissionsNumber;
+                try {
+                    workspaces = workspaceManager.getWorkspaces(user.getId());
+                    List<String> workspaceIds = workspaces.stream()
+                                                          .map(WorkspaceImpl::getId)
+                                                          .collect(Collectors.toList());
+                    permissionsNumber = workspaces.size();
+                    //add workspaces witch are belong to user, but user doesn't have permissions for them.
+                    workspaceManager.getByNamespace(user.getName())
+                                    .stream()
+                                    .filter(workspace -> !workspaceIds.contains(workspace.getId()))
+                                    .forEach(workspaces::add);
+                } catch (ServerException exception) {
+                    reportPrinter.printError("Failed to receive list of related workspaces for user " + user.getId(), auditReport);
+                    continue;
+                }
+                List<AbstractPermissions> wsPermissions = new ArrayList<>();
+                for (WorkspaceImpl workspace : workspaces) {
+                    try {
+                        wsPermissions.add(permissionsManager.get(user.getId(), "workspace", workspace.getId()));
+                    } catch (NotFoundException | ConflictException ignored) {
+                        //Continue printing report, error will be printed in workspace permissions row
+                    }
+                }
+                reportPrinter.printUserInfoWithHisWorkspacesInfo(auditReport, user, workspaces, permissionsNumber, wsPermissions);
```

Comment:
Why should you use permissionsNumber as parameter? ReportPrinter can not use wsPermissions.size()?